src/MessageNames.php

```diff
@@ -13,18 +13,14 @@
 
 namespace PMG\Queue;
 
-/**
- * ABC for messages, implements `getName` as the class name.
- *
- * @since   2015-06-11
- */
-trait MessageTrait
+trait MessageNames
 {
-    /**
-     * @see Message::getName
-     */
-    public function getName()
+    protected static function nameOf(Message $message) : string
```

Comment:
Why `protected static` instead of `private`? Is there ever a benefit to having a `protected` trait method?

Reply:
habit, I guess. I tend to do `protected` in traits when I want those methods to be used outside of the trait itself. While `private` means, to me at least, that the only calls to that method are inside the trait.

Not that its enforceable in PHP itself, though.

Reply:
Interesting argument, but since you can't call `parent::` on a overloaded trait method, I don't think it makes sense.